web3.js/src/publickey.ts

```diff
@@ -54,22 +61,41 @@ export class PublicKey extends Struct {
   constructor(value: PublicKeyInitData) {
     super({});
     if (isPublicKeyData(value)) {
-      this._bn = value._bn;
+      if (typeof value._bn === 'object') {
+        // Legacy implementation of public key storage as a `BN` instance.
+        this._bn = toBigIntLE(value._bn.toBuffer('le', 32));
+      } else {
+        this._bn = value._bn;
+      }
+    } else if (typeof value === 'number') {
+      this._bn = BigInt(value);
+    } else if (typeof value === 'bigint') {
+      this._bn = value;
+      if (
+        this._bn >=
+        // 2^256
+        BigInt(
+          '115792089237316195423570985008687907853269984665640564039457584007913129639936',
+        )
```

Comment:
I generally avoid constants, because they make people have to hop around the code. I avoid them doubly when there's literally only one place they're needed.

wrt the `- 1` idea, I also generally don't want to make the runtime do math when it doesn't have to.

Reply:
Sorry, I meant to define it right there, so no hopping, but it's no big deal. And for -1, I meant encoding the string as '11579...9935` instead, so it would really mean `PUBLIC_KEY_MAXIMUM_VALUE`